app/services/curation_concerns/curation_concern.rb

```diff
@@ -1,9 +1,20 @@
 module CurationConcerns
   module CurationConcern
-    def self.actor(curation_concern, *args)
+    # Returns the top-level actor on the stack
+    def self.actor(curation_concern, current_user, attributes)
+      AddToCollectionActor.new(curation_concern, current_user, attributes,
```

Comment:
If we want this to be more functional we'd just need to have create and save create and return a state object that encapsulates status & parameters.  Would that be more to your liking?